AutomatedTesting/Gem/PythonTests/scripting/ScriptEvents_AllParamDatatypes_CreationSuccess.py

```diff
@@ -138,30 +124,25 @@ async def run_test(self):
 
         # 7) Set different names and datatypes for each parameter
         tools.expand_qt_container_rows(self, NAME_STRING)
-        children = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, NAME_STRING)
         index = 0
-        for child in children:
-            line_edit = child.findChild(QtWidgets.QLineEdit)
-            if line_edit is not None and line_edit.text() == "ParameterName":
-                line_edit.setText(f"param_{index}")
+        name_fields = tools.get_script_event_parameter_name_text(self)
+        for name_field in name_fields:
+            if name_field is not None and name_field.text() == PARAMETER_NAME:
+                name_field.setText(f"param_{index}")
                 index += 1
 
-        children = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, "Type")
         index = 0
-        for child in children:
-            combo_box = child.findChild(QtWidgets.QComboBox)
-            if combo_box is not None and index < N_VAR_TYPES:
-                combo_box.setCurrentIndex(index)
+        type_combo_boxes = tools.get_script_event_parameter_type_combobox(self)
+        for type_combo_box in type_combo_boxes:
+            if type_combo_box is not None and index < N_VAR_TYPES:
+                type_combo_box.setCurrentIndex(index)
                 index += 1
```

Comment:
This should be extracted to a local helper function or to the tools if generalized enough.

Reading through this code is not easily understandable for intent where a helper function can help define that with a meaningful name + the docstring comments.

Reply:
migrated this behavior to local functions to make the overall test script flow more readable.